src/main/java/org/mockito/internal/stubbing/answers/AnswerFunctionalInterfaces.java

```diff
@@ -0,0 +1,326 @@
+package org.mockito.internal.stubbing.answers;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Functional interfaces to make it easy to implement answers in Java 8
+ */
+public class AnswerFunctionalInterfaces {
+    /**
+     * Construct an answer from a two parameter answer interface
+     * @param answer answer interface
+     * @param <T> return type
+     * @param <A> input parameter 1 type
+     * @return a new answer object
+     */
+    public static <T, A> Answer<T> toAnswer(final Answer1<T, A> answer) {
+        return new Answer<T>() {
+            @SuppressWarnings("unchecked")
+            public T answer(InvocationOnMock invocation) throws Throwable {
+                return answer.answer(
+                        (A)invocation.getArguments()[0]);
```

Comment:
`invocation.getArgument(x)` can be used. It doesn't change much and it's probably just a minor optimisation.

It might be better to get the array as a local variable and use it directly... but that's such a minor performance improvement, it's probably better not to do so, just for clarity.